src/alexandria3k/file_xml_cache.py

```diff
@@ -0,0 +1,63 @@
+#
+# Alexandria3k Patent grant bibliographic metadata processing
+# Copyright (C) 2023  Aggelos Margkas
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Cache of read and parsed XML files"""
+
+import xml.etree.ElementTree as ET
+
+
+class FileCache:
+    """Cache the reading of a single concatenated XML file"""
+
+    # pylint: disable=too-few-public-methods
+    parse_counter = 0
+
+    def __init__(self):
+        self.cached_patent_xml_id = None
+        self.cached_data = None
+
+    def read(self, xml_chunk, container_id):
+        """
+        Asserts container_id with cached_patent_xml_id. If matched
```

Comment:
Compares…
Also, please try to avoid describing the function's API based on the internal state.
A clearer description would be: "Return the parsed contents of the specified container id in etree form.
Repeated parsing is avoided through caching."